### DISCUSSION TITLE
Why `ConsumerWidget` docs say it's a `StatelessWidget`?

### DISCUSSION THREAD

**Discussion by matuella** [rrousselGit/riverpod#820] [Q&A] [answered]
It extends `ConsumerStatefulWidget`, which is a `StatefulWidget`.

What is the `flutter_riverpod` rationale to make `flutter` consider it a `StatelessWidget`? By just exposing the `ConsumerWidget.build` method in `_ConsumerState.build`, the performance can be considered the same as exposing a  `StatelessWidget`?

**Comment by rrousselGit** [comment, marked as answer]
There is no difference in performance between `StatelessWidget` and `StatefulWidget

The difference between them is the syntax (one vs two classes). `ConsumerWidget` having one class, it's similar to a `StatelessWidget`

**Reply by matuella** [reply]
Oh, I didn't knew about that. Thanks for clarifying that for me, even though it's unrelated to riverpod itself.
